Fixing up some of the descriptions for rewards (#2)
ipopescu/docs: source/docs/casper/concepts/design/rewards.md

- **Block rewards**: These rewards are received for each proposed block that is finalized. They incentivize timely participation in building the chain.
- **Finality signature rewards**: These rewards are received for collecting finality signatures for each block and generating a finality signature to sign a block. They incentivize the creation, propagation and publication of finality signatures, which is critical in establishing common knowledge of the canonical chain.

In each round, a total reward pool is shared among all participating validators proportionally to their weight, as long as all validators fully participate in the processes that are rewarded by this mechanism. These processes are block creation, signature creation, signature propagation and signature publication as part of block proposals. 

> The total reward pool for a block is determined by the `round_seigniorage_rate` setting in the chainspec. This value, together with current total supply and minimum round length, is used to compute the full allocation of rewards for a particular block. The rate itself is set to result in a given target annual inflation, provided that validators fully participate in the rewardable processes described above.

Reviewer: ```suggestion
The `round_seigniorage_rate` setting in the chainspec determines the total reward pool for a block. This value, along with the current total supply and minimum round length, is used to compute the full allocation of rewards for a particular block. The rate itself is set to result in a target annual inflation, provided validators fully participate in the rewardable processes described above.
```

> Each switch block triggers a reward calculation. To account for potential network lag delaying the timely arrival of signatures for finalized blocks, the calculation "looks back" into previous eras. The number of prior blocks to look up is specified using the `signature_rewards_max_delay` setting in the chainspec.

Reviewer: Since we are talking about switch blocks here, emphasize that switch blocks can be rewarded now.

```suggestion
Each switch block triggers a reward calculation. To account for potential network lag delaying the timely arrival of signatures for finalized blocks, the calculation "looks back" into previous eras. In particular, this enables rewards for switch blocks, which was impossible with the prior Highway-specific calculation. The number of prior blocks to look up is specified using the `signature_rewards_max_delay` setting in the chainspec.
```

The following steps summarize the rewards distribution mechanism.

Each round has a reward pool calculated from the `round_seigniorage_rate` chainspec parameter and the current total supply for the relevant era.

In each round, the reward pool is split into two parts for block proposals and finality signature rewards, based on the `finality_signature_proportion` chainspec parameter. 

The amount allocated for finality signatures is split further into two parts: creating and publishing finality signatures. The split is configurable in the chainspec using the `finders_fee` chainspec parameter. 

For each finality signature:
- The creator gets a portion of what was allocated for creating signatures, in proportion to relative consensus weight.

> - The block proposer gets a portion of corresponding to the `finders_fee` chainspec parameter, scaled by total collected signature creator weight divided by total weight in the relevant era.

Reviewer: ```suggestion
- The block proposer gets a portion corresponding to the `finders_fee` chainspec parameter, scaled by the total collected signature creator weight divided by the total weight in the relevant era.
```

<p align="center">
<img src={"/image/design/rewards-pot.png"} alt="Pie chart showing how rewards are split" width="300"/>
</p>

> The rewards calculation takes place at the end of an era. In addition to rewarding everything in that era, the algorithm also looks back into previous era(s), depending on the `signature_rewards_max_delay` parameter, to compensate for delay in creation and distribution of finality signature. In particular, this enables us to reward switch blocks, which was impossible with the prior, Highway-specific calculation.

Reviewer: I think we need to say previous "blocks" (not eras) here. I had eras before, but it wasn't correct.
Also, I'd move the last sentence higher, when introducing the look back mechanism.

```suggestion
The rewards calculation takes place at the end of an era. In addition to rewarding everything in that era, the algorithm also looks back into blocks from the previous era, depending on the `signature_rewards_max_delay` parameter, to compensate for the delay in creating and distributing finality signatures.
```

Author: Maybe it should just say "era" pending clarification from Bart on whether we allow to look back more than one era. But it's correct in principle. We look back _into_ a previous era, but look back _at_ prior blocks.

Reviewer: Q&A from Bart:
Is it possible to look back into multiple eras in the past?
Technically it would be possible, but it would require some eras to be shorter than the lookback window, which is something like 3 or 5 blocks, so it shouldn't happen under normal operation. (We have single-block eras after every upgrade, but we don't look back across an upgrade boundary, so this case is excluded.)

As a result, I've made this update: "the algorithm also looks back into blocks from the previous era".

I think we can go with this to close your PR and then send the main content to Bart for final review.
